### PR TITLE
EVA-1637 vcf validator and asm checker can't uncompress bgzip files properly

### PR DESCRIPTION
**Pull request by jmmut**
History of this issue:
- I generated a **b**gzipped VCF that passes validation if it's decompressed before calling the validator (`zcat file.vcf.gz | vcf_validator`), but doesn't pass validation if the validator itself decompresses the file (`vcf_validator -i file.vcf.gz`). If the file is decompressed and gzipped, it also works. It seems that depending on the block size, the program didn't read properly all lines.
- A test was added to process this file (test/input_files/v4.3/compressed_files/readable/passed/boost_55_can_not_decompress_last_line_of_this_bgzipped.vcf.gz).
- Confirmed with the bgzip people at 
https://github.com/samtools/htslib/issues/953 that bgzip should be correctly read by any gzip reader. We found out that newer versions of boost decompress the offending file correctly.
- Changed the build in linux to download and compile Boost, instead of installing from distro packages.
  - [x] Changed the install_dependencies.sh script for manual installation.
  - [x] Changed the travis linux build.
  - [ ] Changed the docker build. NOT DONE: it seems this failed even in master, so it's out of the scope of this ticket. The docker build is removed for now.
  - [x] Changed README to reflect the new (and easier) installation steps.
- The OSX validator passes the test without modification, so the build was not changed for OSX.
- Decompression for Windows has never worked and is low priority, so no changes there.
- The binaries from travis for the release will be compiled with gcc 5 (the xenial default). The travis build for gcc 4.8 and gcc 6 were removed because they were not working, possibly due to ABI incompatibilities with ODB. Maybe compiling all dependencies with the same ABI fixes this, but that's out of scope of this ticket.
- The locale problem https://github.com/EBIvariation/vcf-validator/issues/184 prevents from building in Ubuntu 18, but the precompiled binary (built in Ubuntu 16) can be run without problems in Ubuntu 18, so that issue is half done.

The cause of https://github.com/EBIvariation/vcf-validator/issues/190 wasn't identified, but this PR maybe fixes it, as the validator now reads correctly bgzip files.

### CODE REVIEW

File: install_dependencies.sh [hunk 1]
```diff
@@ -68,27 +71,39 @@ mkdir -p $dependencies_dir && cd $dependencies_dir
 
 echo "installing libodb"
 wget http://codesynthesis.com/download/odb/2.4/libodb-2.4.0.tar.bz2 -O /tmp/libodb.tar.bz2
-tar jxvf /tmp/libodb.tar.bz2
-cd libodb-2.4.0 && ./configure && make && cd ..
+tar jxf /tmp/libodb.tar.bz2
+cd libodb-2.4.0 && ./configure && make
+RESULT=$?
+INSTALL_FAILED=$(( $INSTALL_FAILED + $RESULT ))
```
Comment (sundarvenkata-EBI) [inline install_dependencies.sh]: Will a bitwise-or be better here? I am guessing bash won't return negative exit codes, so we are covered here, right?

File: install_dependencies.sh [hunk 2]
```diff
@@ -19,6 +19,7 @@ it installs the given dependencies:
   - curl library                            curl-7.62.0
   - openssl library                         openssl-1.1.1a
   - c-ares library                          c-ares-1.15.0
+  - boost library                           boost-1.72.0
 
 for linux:
 ./install_dependencies.sh linux
```
Comment (sundarvenkata-EBI) [inline install_dependencies.sh]: Are elevated permissions needed for running this script (make outputs etc.,)? In that case, should we call out this script to be executed as a privileged user?

Comment (sundarvenkata-EBI) [reply install_dependencies.sh]: Never mind, looks like the make install commands seem to target the local directory anyway...

File: install_dependencies.sh (hunk 1 above)
Comment (jmmut) [reply install_dependencies.sh]: please suggest a syntax because I can't make the "or" work.

Comment (jmmut) [reply install_dependencies.sh]: actually, I realised there's a `set -e` at the top that will make the script exit on the first failure, so I will leave only the "everything ok" at the end, as the "install failed" message will never show

**Pull request merged**